packages/react/slot/src/Slot.stories.tsx

```diff
@@ -15,6 +15,44 @@ export const WithSlottable = () => (
   </SlotWithSlottable>
 );
 
+export const WithComposedEvents = () => (
+  <>
+    <h1>Should log both</h1>
+    <SlotWithPreventableEvent>
+      <button onClick={() => console.log('button click')}>Slot event not prevented</button>
+    </SlotWithPreventableEvent>
+
+    <h1>Should log "button click"</h1>
+    <SlotWithPreventableEvent>
+      <button
+        onClick={(event) => {
+          console.log('button click');
+          event.preventDefault();
+        }}
+      >
+        Slot event prevented
+      </button>
+    </SlotWithPreventableEvent>
+
+    <h1>Should log both</h1>
+    <SlotWithoutPreventableEvent>
+      <button onClick={() => console.log('button click')}>Slot event not prevented</button>
+    </SlotWithoutPreventableEvent>
+
+    <h1>Should log both</h1>
+    <SlotWithoutPreventableEvent>
+      <button
+        onClick={(event) => {
+          console.log('button click');
+          event.preventDefault();
+        }}
+      >
+        Slot event prevented
+      </button>
+    </SlotWithoutPreventableEvent>
+  </>
```

Comment:
If we fixed [this issue](https://github.com/radix-ui/primitives/pull/890/files#r722090555) then this story wouldn't have worked which is why I think the `Slot` shouldn't handle it.